src/ViewModeResolver.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Drupal\format_strawberryfield;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Cache\UseCacheBackendTrait;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\strawberryfield\StrawberryfieldUtilityServiceInterface;
+
+/**
+ * View Mode resolver service.
+ *
+ * @ingroup format_strawberryfield
+ */
+class ViewModeResolver implements ViewModeResolverInterface {
+  use UseCacheBackendTrait;
+
+  /**
+   * The config factory service.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   */
+  protected $configFactory;
+
+  /**
+   * The Strawberry Field Utility Service
+   *
+   * @var \Drupal\strawberryfield\StrawberryfieldUtilityService
+   */
+  protected $strawberryfieldUtility;
+
+  /**
+   * DisplayResolver constructor.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   * @param \Drupal\strawberryfield\StrawberryfieldUtilityServiceInterface $strawberryfield_utility_service
+   *   The SBF utility Service
+   */
+  public function __construct(StrawberryfieldUtilityServiceInterface $strawberryfield_utility_service, ConfigFactoryInterface $config_factory) {
+    $this->strawberryfieldUtility = $strawberryfield_utility_service;
+    $this->configFactory = $config_factory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCandidates(ContentEntityInterface $entity) {
+    if ($ado_types = $this->getADOTypes($entity)) {
+      $view_modes = array_filter($this->getSortedMappings(), function($mapping) use ($ado_types) {
+        return ($mapping['active'] == TRUE) && in_array($mapping['jsontype'], $ado_types);
+      });
+      return $view_modes;
+    }
+
+    return ['default'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function get(ContentEntityInterface $entity) {
+    $ado_types = $this->getADOTypes($entity);
+    if (!empty($ado_types)) {
+      foreach ($this->getSortedMappings() as $mapping) {
+        if (($mapping['active'] == TRUE) && in_array($mapping['jsontype'], $ado_types)) {
+          return $mapping['view_mode'];
+        }
+      }
+    }
+
+    return 'default';
```

Comment:
Is default the right one? Where are the saved "view modes" stored when a user selects instead of the default (during edit/create) one?

Reply:
I'm afraid that comes from display suite, `$node->ds_switch->value`, and I'm not sure we want that dependency. Something like `\Drupal::service('entity_display.repository')->getViewDisplay($entity_type, $bundle, $view_mode);` will get us the default bundle display, but I'm not sure if it's enough.